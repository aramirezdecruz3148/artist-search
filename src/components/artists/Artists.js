import React from 'react';
import PropTypes from 'prop-types';
import Artist from './Artist';

function Artists({ artistArray }) {
  const artistList = artistArray.map(({ artistId, artistName, artistDeets }) => (
    <li key={artistId}>
      <Artist
        artistId={artistId}
        artistName={artistName}
        artistDeets={artistDeets}
      />
    </li>
  ));

  return (
    <ul>
      {artistList}
    </ul>
  );
}

Artists.propTypes = {
  artistArray: PropTypes.arrayOf(PropTypes.shape({
    artistId: PropTypes.string.isRequired,
    artistName: PropTypes.string.isRequired,
    artistDeets: PropTypes.string
  })).isRequired
};

export default Artists;
