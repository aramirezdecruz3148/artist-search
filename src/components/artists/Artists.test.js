import React from 'react';
import { shallow } from 'enzyme';
import Artists from './Artists';

describe('Artists component', () => {
  it('renders Artists', () => {
    const wrapper = shallow(<Artists artistArray={[{ artistId: '1231', artistName: 'name', artistDeets: 'they do stuff' }, { artistId: 'uiiuhh', artistName: 'someone', artistDeets: 'they did stuff too' }]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
