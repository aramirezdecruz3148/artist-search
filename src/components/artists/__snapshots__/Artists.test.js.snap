// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artists component renders Artists 1`] = `
<ul>
  <li
    key="1231"
  >
    <Artist
      artistDeets="they do stuff"
      artistId="1231"
      artistName="name"
    />
  </li>
  <li
    key="uiiuhh"
  >
    <Artist
      artistDeets="they did stuff too"
      artistId="uiiuhh"
      artistName="someone"
    />
  </li>
</ul>
`;
